diff --git a/src/components/oss/Header.vue b/src/components/oss/Header.vue
index ce6d73bb1d41620526299a403df73a41ee25ac9d..af8ad4fe7f45587021b5058587b65bd1ca65c903 100644
--- a/src/components/oss/Header.vue
+++ b/src/components/oss/Header.vue
@@ -197,8 +197,10 @@ onUnmounted(() => {
       <!-- Left side: Logo + Nav -->
       <div class="flex items-center gap-10">
         <a href="/" class="shrink-0">
+          <slot name="nav-bar-title-before" />
           <img class="h-4 block dark:hidden" :src="logoDark" :alt="logoAlt" />
           <img class="h-4 hidden dark:block" :src="logoLight" :alt="logoAlt" />
+          <slot name="nav-bar-title-after" />
         </a>
 
         <!-- Desktop navigation -->
diff --git a/src/components/vitepress-default/Layout.vue b/src/components/vitepress-default/Layout.vue
index 80fe5ed1b186546c49707a518d5f6f0cd04f5edc..4e5b8120bbe2ac362bea3d14893a41dbef51e003 100644
--- a/src/components/vitepress-default/Layout.vue
+++ b/src/components/vitepress-default/Layout.vue
@@ -34,7 +34,10 @@ provide(layoutInfoInjectionKey, { heroImageSlotExists })
     <slot name="layout-top" />
     <VPSkipLink />
     <VPBackdrop class="backdrop" :show="isSidebarOpen" @click="closeSidebar" />
-    <OSSHeader />
+    <OSSHeader>
+      <template #nav-bar-title-before><slot name="nav-bar-title-before" /></template>
+      <template #nav-bar-title-after><slot name="nav-bar-title-after" /></template>
+    </OSSHeader>
 
     <div class="flex flex-col min-h-screen">
       <!-- Content wrapper with borders -->
diff --git a/src/layouts/VPLayout.vue b/src/layouts/VPLayout.vue
index 9b44852a4f6cceea02b4eb446cdc9174c200c6ae..1f28234424d3064ff234a7386974f8c0a7ca1abc 100644
--- a/src/layouts/VPLayout.vue
+++ b/src/layouts/VPLayout.vue
@@ -44,7 +44,10 @@ const useVPLayout = computed(() => {
   </div>
   <div v-else class="marketing-layout" :data-theme="frontmatter.theme" :data-variant="variant">
     <TopBanner />
-    <OSSHeader />
+    <OSSHeader>
+      <template #nav-bar-title-before><slot name="nav-bar-title-before" /></template>
+      <template #nav-bar-title-after><slot name="nav-bar-title-after" /></template>
+    </OSSHeader>
     <Content />
   </div>
 </template>
